src/libstd/process.rs

```diff
@@ -875,10 +875,23 @@ impl Child {
 ///
 /// # Examples
 ///
+/// Due to this function’s behaviour regarding destructors, a conventional way
+/// to use the function is to extract the actual computation to another
+/// function and compute exit code from its return value:
+///
 /// ```
-/// use std::process;
+/// use std::io::Write;
+/// fn run_app() -> Result<(), ()> { Ok(()) }
 ///
-/// process::exit(0);
+/// fn main() {
+///     ::std::process::exit(match run_app() {
+///        Ok(_) => 0,
+///        Err(err) => {
+///            writeln!(::std::io::stderr(), "error: {:?}", err).unwrap();
+///            1
+///        }
+///     });
+/// }
 /// ```
```

Comment:
I personally prefer to not include an import for only one use of a function. I incorporated some of the other changes.